plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -2398,6 +2414,99 @@ else if (isMetadataColumnId(columnHandle.getId())) {
                 false));
     }
 
+    @Override
+    public Optional<AggregationApplicationResult<ConnectorTableHandle>> applyAggregation(
+            ConnectorSession session,
+            ConnectorTableHandle handle,
+            List<AggregateFunction> aggregates,
+            Map<String, ColumnHandle> assignments,
+            List<List<ColumnHandle>> groupingSets)
+    {
+        IcebergTableHandle tableHandle = (IcebergTableHandle) handle;
+
+        // Iceberg's metadata cannot be used for aggregation calculation.
+        // As equality deletes do not reflect at the metadata/count level.
+        if (hasEqualityDeletes(session, tableHandle)) {
+            return Optional.empty();
+        }
+
+        if (!isAggregationPushdownEnabled(session)) {
+            return Optional.empty();
+        }
+
+        // not supporting unenforced predicate
+        if (!tableHandle.getUnenforcedPredicate().isNone()
+                && tableHandle.getUnenforcedPredicate().getDomains().isPresent()
+                && !tableHandle.getUnenforcedPredicate().getDomains().get().isEmpty()) {
+            return Optional.empty();
+        }
+
+        // not supporting group by
```

Comment:
This is pretty limited. I think it would make sense to support simple grouping sets in the first iteration to future-proof the design.